### ISSUE TITLE
Generated migration fail when using `strong_migrations`

### ISSUE THREAD

**Issue by BrentWheeldon** [ezcater/sidekiq_publisher#18]
When running the generated migration (`CreateNotifyOnSidekiqPublisherJobsInsertTrigger`) an error is generated if `strong_migrations` is in place.

```
== 20180725122958 CreateNotifyOnSidekiqPublisherJobsInsertTrigger: migrating ==
-- set_setting(:lock_timeout, 750)
-- set_setting(:statement_timeout, 1500)
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

 __          __     _____ _______ _
 \ \        / /\   |_   _|__   __| |
  \ \  /\  / /  \    | |    | |  | |
   \ \/  \/ / /\ \   | |    | |  | |
    \  /\  / ____ \ _| |_   | |  |_|
     \/  \/_/    \_\_____|  |_|  (_)  #strong_migrations

The strong_migrations gem does not support inspecting what happens inside an
execute call, so cannot help you here. Please make really sure that what
you're doing is safe before proceeding, then wrap it in a safety_assured { ... } block.

/usr/src/app/db/migrate/20180725122958_create_notify_on_sidekiq_publisher_jobs_insert_trigger.rb:7:in `up'
```

I was thinking of changing the generator to check `defined? StrongMigrations` and wrap the `execute` calls in `safely_assured` blocks - if that's cool with y'all I'll open a PR.

**Comment by tjwp** [comment]
I think this is actually due to an issue with the activerecord-postgres_pub_sub gem that I already addressed: https://github.com/ezcater/activerecord-postgres_pub_sub/pull/8

If that gem had been updated too then the generated migration would have been "safe".

I can add a version restriction in `sidekiq_publisher` to require the later version, but I think this app was in the unique position of already having `activerecord-postgres_pub_sub` due to `ezcater-kafka`, but not yet including the `sidekiq_publisher`. Less likely to arise for new apps.
